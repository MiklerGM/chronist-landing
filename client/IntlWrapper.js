import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
