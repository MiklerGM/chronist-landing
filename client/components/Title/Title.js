import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import './Title.less';

const TitleImageRu = require('./logo-grey-ru.svg');
const TitleImageEn = require('./logo-grey-en.svg');

const Title = ({ locale }) => {
  let TitleImage = {};
  if (locale === 'ru') {
    TitleImage = TitleImageRu;
  } else {
    TitleImage = TitleImageEn;
  }
  return (
    <div id="title">
      <div className='text-center'>
        <img src={TitleImage} draggable="false" alt='Chronist logo' />
        <h4>
          <FormattedMessage
            id='home.title'
          />
        </h4>
      </div>
    </div>
  );
};

Title.propTypes = {
  locale: PropTypes.string.isRequired
};

export default Title;
