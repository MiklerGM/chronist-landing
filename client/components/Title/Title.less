@import '../../styles/colors.less';

#title {
  margin: 0 auto;
  background: linear-gradient(rgba(0, 0, 0, 0.1), rgba(0, 0, 0, 0.1)), url("../../images/back-progressive.png");
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-position: center;
  background-size: cover;
  padding-top: 40px;
  padding-bottom: 120px;

  & img {
    min-width: 65%;
    user-select: none;
    margin-bottom: 10px;

    &::selection {
      background-color: transparent;
      color: #000;
    }
    @media screen and (max-width: 600px) { min-width: 85%; }
    @media screen and (min-width: 1000px) { min-width: 700px; }
  }

  h4 {
    margin: 0 auto;
    max-width: 65%;
    line-height: 1.5em;
    font-weight: 400;
    text-align: justify;
    color: @white;

    @media screen and (max-width: 600px) { max-width: 85%; }

    @media screen and (min-width: 1000px) { max-width: 700px; }
  }
}
