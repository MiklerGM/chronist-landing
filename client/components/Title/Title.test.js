import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../../IntlWrapper';
import Title from './Title';

const TestComponent = () => (
  <IntlWrapper>
    <Title locale='en' />
  </IntlWrapper>
);

// it must render
it('Title render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('Title render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('Title Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
