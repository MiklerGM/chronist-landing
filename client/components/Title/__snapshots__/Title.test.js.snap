// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Snapshot 1`] = `
<div
  id="title"
>
  <div
    className="text-center"
  >
    <img
      alt="Chronist logo"
      draggable="false"
      src="test-file-stub"
    />
    <h4>
      <span>
        Мы хотим понимать как устроен мир. Находить причинно-следственные связи между событиями. Иметь возможность заглянуть в историю на своих условиях, без толстых пыльных учебников и хрупких карт за стёклами музеев. Мы объединили исторические карты с последовательностью событий и добавили немного интерактивности.
      </span>
    </h4>
  </div>
</div>
`;
